modules/op-setting-up-openshift-pipelines-to-view-project-vulnerabilities.adoc

```diff
@@ -0,0 +1,137 @@
+kind: Task
+metadata:
+  name: vulnerability-scan  # <.>
+  annotations:
+    task.output.location: results  # <.>
+    task.results.format: application/json
+    task.results.key: SCAN_OUTPUT  # <.>
+spec:
+  results:
+    - description: CVE result format  # <.>
+      name: SCAN_OUTPUT
+      type: string
+  steps:
+    - name: roxctl  # <.>
+      image: quay.io/roxctl-tool-image  # <.>
+      env:
+        - name: ENV_VAR_NAME_1  # <.>
+          valueFrom:
+            secretKeyRef:
+              key: secret_key_1            
+              name: secret_name_1
+      env:
+        - name: ENV_VAR_NAME_2
+          valueFrom:
+            secretKeyRef:
+              key: secret_key_2            
+              name: secret_name_2
+      script: | # <.>
+        #!/bin/sh
+        # Sample shell script
+        echo "ENV_VAR_NAME_1: " $ENV_VAR_NAME_1
+        echo "ENV_VAR_NAME_2: " $ENV_VAR_NAME_2
+        jq --version
+        curl -k -L -H "Authorization: Bearer $ENV_VAR_NAME_1" https://$ENV_VAR_NAME_2/api/cli/download/roxctl-linux --output ./roxctl
+        chmod +x ./roxctl 
+        echo "roxctl version"
+        ./roxctl version
+        echo "image from pipeline: " 
+        
+        # Replace the following line with your dynamic image logic
+        DYNAMIC_IMAGE=$(get_dynamic_image_logic_here)
+        echo "Dynamic image: $DYNAMIC_IMAGE"
+        ./roxctl image scan --insecure-skip-tls-verify -e $ENV_VAR_NAME_2 --image $DYNAMIC_IMAGE --output json  > roxctl_output.json
+        more roxctl_output.json
+        jq -rce \  # <.>
+          '{vulnerabilities:{
+          critical: (.result.summary.CRITICAL),
+          high: (.result.summary.IMPORTANT),
+          medium: (.result.summary.MODERATE),
+          low: (.result.summary.LOW)
+            }}' scan_output.json | tee $(results.SCAN_OUTPUT.path)
+----
+<.> The name of your task.
+<.> The location for storing the task outputs.
+<.> The naming convention of the scan task result. A valid naming convention must end with the `SCAN_OUTPUT` string. For example, SCAN_OUTPUT, MY_CUSTOM_SCAN_OUTPUT, or ACS_SCAN_OUTPUT.
+<.> The description of the result.
+<.> The name of the vulnerability scanning tool that you have used. 
+<.> The location of the actual image containing the scan tool.
+<.> The tool-specific environment variables.
+<.> The shell script to be executed with json output. For example, scan_output.json.
+<.> The format to extract vulnerability summary (adjust `jq` command for different JSON structures).
+
+. Update an appropriate Pipeline to add vulnerabilities specifications in the following format:
+
++
+[source,yaml]
+----
+...
+spec:
+  results:
+    - description: The common vulnerabilities and exposures (CVE) result
+      name: SCAN_OUTPUT
+      type: $(tasks.vulnerability-scan.results.SCAN_OUTPUT)
+----
+
+.Verification
+
+* Navigate to the PipelineRun details page and review the Vulnerabilities row for a visual representation of identified vulnerabilities.
```

Comment:
I think Vulnerabilities should be in bold if it is a UI element (applies in other places where it is mentioned to)